### ISSUE TITLE
Code coverage report for an address that contains only a contract interface

### ISSUE THREAD

**Issue by m-Peter** [onflow/cadence#2361]
### Current Behavior

Generating code coverage report for an address/location that contains only a contract interface, or generally Cadence source code that is not meant to be interpreted/executed, returns a `percentage` of `"NaN%"`. E.g:

![coverage-for-resource-interface](https://user-images.githubusercontent.com/1778965/222155196-7298190f-3844-4d02-9c9a-f41ea3f11f49.png)

### Expected Behavior

How do we handle such a case? Should we show a percentage of `100%` ?

### Steps To Reproduce

This happens when adding code coverage in `flow-emulator`, and expose it in the REST API.

### Environment

```markdown
- Cadence version: v0.34.0
- Network: Emulator
```

**Comment by bluesign** [comment]
@m-Peter 

Those are pre/post conditions ( `ast.Condition` ), I think you should add that to program inspector. 

Conditions are taking `expression` instead of statement.

**Comment by m-Peter** [comment]
@bluesign Great catch :100: :100: :100: 

I checked every line hit, and indeed 7/8 are in fact pre/post conditions. However, the line `0` which, has `14` hits, remains a mystery.

**Comment by bluesign** [comment]
@m-Peter that one is a bug I guess, I ran into that when improving debugger too.

**Comment by m-Peter** [comment]
I see, I wasn't able to identify where it comes from. However, I did find a way to include pre/post conditions for all functions, whether they are defined in contract interfaces or in concrete contracts.

![Screenshot from 2023-03-04 14-39-58](https://user-images.githubusercontent.com/1778965/222903516-b773ff0f-04bb-4fef-9403-c0ac0e5160ce.png)

**Comment by bluesign** [comment]
you can ignore probably line 0 anyway. normally line number starts with 1 I guess

**Comment by m-Peter** [comment]
That makes total sense :+1:

**Comment by turbolent** [comment]
@m-Peter Do you have an example program where you see reports of line 0?

**Comment by bluesign** [comment]
> @m-Peter Do you have an example program where you see reports of line 0?

beforeStatements iirc

**Comment by m-Peter** [comment]
I have a branch on `flow-cli`, where I am adding support for code coverage there as well. This peculiar case manifests only for the `FungibleToken` contract interface, and it is visible only by checking the location key format.

**Comment by m-Peter** [comment]
```go
if location.String() == "ee82856bf20e2aa6.FungibleToken" {
	if statement.StartPosition().Line == 0 {
		fmt.Printf("Element is %T, Value is %v\n", statement, statement)
	}
}
```
I added the above check, on `newOnStatementHandler()`. This is what is printed out:

```cadence
Element is *ast.VariableDeclaration, Value is var exp1 = self.balance
Element is *ast.VariableDeclaration, Value is var exp2 = from.balance
Element is *ast.VariableDeclaration, Value is var exp0 = self.balance
```

Not sure if it helps somehow.

**Comment by turbolent** [comment]
@m-Peter Thanks for the bug report! I fixed this in #2373
